test/WebSites/ModelBindingWebSite/Controllers/ValidationController.cs

```diff
@@ -21,6 +22,51 @@ public bool AvoidRecursive(SelfishPerson selfishPerson)
         {
             return ModelState.IsValid;
         }
+
+        public IActionResult RequiredAttributeOnValueAndReferenceTypes([FromBody] Rectangle drawing)
```

Comment:
lets keep the action names more realistic(at least easy to read), the test names can depict what is being tested... I really like the models you selected and the way this is coded up, we could add comments if you want to explicitly clarify what is being tested here.